maxim/maxim.py

```diff
@@ -13,16 +13,12 @@
 from .blocks.misc_gating import CrossGatingBlock
 from .blocks.others import UpSampleRatio
 from .blocks.unet import UNetDecoderBlock, UNetEncoderBlock
-from .layers import Resizing
+from .layers import ResizingDown
 
 Conv1x1 = functools.partial(layers.Conv2D, kernel_size=(1, 1), padding="same")
 Conv3x3 = functools.partial(layers.Conv2D, kernel_size=(3, 3), padding="same")
-ConvT_up = functools.partial(
-    layers.Conv2DTranspose, kernel_size=(2, 2), strides=(2, 2), padding="same"
-)
-Conv_down = functools.partial(
-    layers.Conv2D, kernel_size=(4, 4), strides=(2, 2), padding="same"
-)
+ConvT_up = functools.partial(layers.Conv2DTranspose, kernel_size=(2, 2), strides=(2, 2), padding="same")
+Conv_down = functools.partial(layers.Conv2D, kernel_size=(4, 4), strides=(2, 2), padding="same")
 
 
 def MAXIM(
```

Comment:
What was the main change here needed to facilitate `(None, None, 3)` input tensors?

Reply:
Adding the resize layers and piping ratio to them.

Reply:
Resizing layers were there previously too. Do you mean having `ratio` instead of separate `height` and `width` was the key change?

Reply:
To this specific file (maxim.py) yes, for the PR no.
The key changes were to rewrite the einops operations in pure TF.
A secondary change was to make [dim_u](https://github.com/LeviVasconcelos/maxim-tf/blob/main/maxim/blocks/misc_gating.py#L121) and [dim_v](https://github.com/LeviVasconcelos/maxim-tf/blob/main/maxim/blocks/misc_gating.py#L130) independent of the input image size.

Reply:
> This casting to int() is premature, since some of those values are supposed to be < 0, and casting to int will project them to 0.

Would appreciate an example.

Reply:
Consider [this](https://github.com/sayakpaul/maxim-tf/blob/feat/dynamic-shape/maxim/maxim.py#L267) and [this](https://github.com/sayakpaul/maxim-tf/blob/feat/dynamic-shape/maxim/maxim.py#L224) lines of code from the feat/dynamic-shape branch.

If you have j > i on the second one, you get a below zero fraction which will be projected to zero.
Same goes for the same line: whenever (depth - j - 1 - i) is positive, you have a below zero ratio. 
If I recall correctly, the second case (depth - j - 1 - i) was yielding 0.25, 0.5, and so on. Thus I changed it.

The way i did to overcome this, was to actually pass the float number, and compute the new desired image size: img_size * ratio. And just after that converting back to int.